### PR TITLE
Update CDI version to 1.21

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:
Update the CDI version to 1.21. So a new API changes (the new value of DataVolumePhase: WaitForFirstConsumer) can be used. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,44 +34,42 @@ require (
...
+	k8s.io/api v0.18.2
+	k8s.io/apiextensions-apiserver v0.18.2
+	k8s.io/apimachinery v0.18.2
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/kube-aggregator v0.16.4
 	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a
 	k8s.io/utils v0.0.0-20190801114015-581e00157fb1
 	kubevirt.io/client-go v0.0.0-00010101000000-000000000000
-	kubevirt.io/containerized-data-importer v1.10.9
+	kubevirt.io/containerized-data-importer v1.21.0
```
Comment (mhenriks) [inline go.mod:70]: Need to update here as well:  https://github.com/kubevirt/kubevirt/blob/master/staging/src/kubevirt.io/client-go/go.mod#L28

Comment (mhenriks) [reply go.mod:70]: and make sure to make `make generate` as well

**Comment by awels** [comment]
@dhiller Looks like something is going wrong with travis here, you made some changes there recently, should we just rebase and resubmit this?

**Comment by brybacki** [comment]
I have triggered travis manually and again it is green, but is still not reporting
https://travis-ci.org/github/kubevirt/kubevirt/builds/719894303

**Comment by brybacki** [comment]
BTW: in the end this PR is almost the same as https://github.com/kubevirt/kubevirt/pull/3931, but this time we want this merged

**Comment by dhiller** [comment]
> @dhiller Looks like something is going wrong with travis here, you made some changes there recently, should we just rebase and resubmit this?

We had some problems after moving coverage to prow, fix is in the works: #4065

**Comment by awels** [comment]
Yeah we would like to get this is some time soon as we have another PR that is dependent on this one. @brybacki looks like you need to rebase due to some conflicts.

**Comment by brybacki** [comment]
Is this test a flaky test?:
[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system] VM Live Migration Starting a VirtualMachineInstance migration monitor Should detect a failed migration expand_more

/test pull-kubevirt-e2e-k8s-1.17

**Comment by brybacki** [comment]
now this one:
[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachine VM rename VM update [test_id:4659]should succeed expand_less	
/test pull-kubevirt-e2e-k8s-1.17

**Comment by brybacki** [comment]
We need this version bump to use new API/new phase for pr #4025, 
David, I have found you already commented on #3931 which was almost the same ;)
/assign davidvossel

**Comment by brybacki** [comment]
/test pull-kubevirt-e2e-k8s-1.18


looks like a flake:
[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:component]VMIlifecycle [rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:component]Creating a VirtualMachineInstance with node tainted [test_id:1635]the vmi with tolerations should be scheduled expand_less
--

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Comment by brybacki** [comment]
@davidvossel I needed to rebase, to satisfy pull-kubevirt-generate, and bot removed the lgtm

**Pull request merged**